plugin/db/mysql/restore.go

```diff
@@ -263,37 +264,33 @@ func (driver *Driver) RestoreBackupToPITRDatabase(ctx context.Context, backup io
 }
 
 // GetBinlogReplayList returns the path list of the binlog that need be replayed.
-func GetBinlogReplayList(startBinlogInfo api.BinlogInfo, binlogDir string) ([]string, error) {
+func GetBinlogReplayList(startBinlogInfo, targetBinlogInfo api.BinlogInfo, binlogDir string) ([]string, error) {
 	startBinlogSeq, err := GetBinlogNameSeq(startBinlogInfo.FileName)
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot parse the start binlog file name %q", startBinlogInfo.FileName)
 	}
+	targetBinlogSeq, err := GetBinlogNameSeq(targetBinlogInfo.FileName)
+	if err != nil {
+		return nil, errors.Wrapf(err, "cannot parse the target binlog file name %q", targetBinlogInfo.FileName)
+	}
 
 	metaList, err := getSortedLocalBinlogFilesMeta(binlogDir)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to read local binlog metadata files from directory %s", binlogDir)
 	}
 
-	startIndex := -1
-	for i, meta := range metaList {
-		if meta.seq >= startBinlogSeq {
-			startIndex = i
-			break
-		}
-	}
-	if startIndex == -1 {
-		log.Error("No binlog files found locally after given start binlog info", zap.Any("startBinlogInfo", startBinlogInfo))
-		return nil, errors.Errorf("no binlog files found locally after given start binlog info: %v", startBinlogInfo)
+	startIndex, err := findBinlogSeqIndex(metaList, startBinlogSeq)
```

Comment:
"findBinlogSeqIndex() and metaList[startIndex : targetIndex+1]" should be handled by a standalone function with unit test.

Reply:
Done.